lib/src/locator.dart

```diff
@@ -34,14 +35,15 @@ final GetIt locator = GetIt.instance;
 
 void setupLocator() {
   locator
-    ..registerLazySingleton<Logger>(Logger.new)
+    ..registerSingletonAsync<Logger>(() async => Logger())
 
     //----------------------//
     // ----- Services ----- //
     //----------------------//
-    ..registerSingletonAsync<AuthService>(() async => AuthService())
+    ..registerSingletonWithDependencies<AuthService>(AuthService.new,
+        dependsOn: [Logger])
     ..registerSingleton<DialogService>(DialogService())
-    ..registerLazySingleton<PreferencesService>(PreferencesService.new)
+    ..registerSingleton<PreferencesService>(PreferencesService())
```

Comment:
Same question, why switching to Singleton?

Reply:
Change reverted